### PR TITLE
refactor to make the common code be a setup function

### LINKED ISSUE #468: Cannot use debug with Rollup
There're a troubles both using `debug` directly (via `import debug from 'debug'`) and as a dependency to another package, with Rollup bundler.

1. `import debug from 'debug'`:
> Non-existent export 'default' is imported from node_modules/debug/src/browser.js

I believe this is because of [that dynamic stuff](https://github.com/visionmedia/debug/blob/52e1f21284322f167839e5d3a60f635c8b2dc842/src/browser.js#L7-L11).

2. `import SocketIo from 'socket.io-client'` which has `debug` as a dependency:
> 'default' is not exported by 'node_modules/socket.io-client/node_modules/debug/src/browser.js'

As far as I understood, the main idea behind `browser.js` is to import kinda «base» debug object and upgrade it with browser-specific methods' implementations. I believe this can be rewritten in more static and bundler-friendly way. Like requiring base, propagate it with methods and then export as a default in different statements which would simplify the analysis by Rollup and other-like.

I'm not an active `debug` user, but `debug` is very popular tool, so many of my packages' of choice using it as a dependency. So I stumble upon it when bundling, from time to time. Leveraging any workarounds is diffcult, since I do not have precision control over dependencies of my dependencies.

### PR DESCRIPTION
**Pull request by TooTallNate**
I believe this will fix #468.

It also enables us to be able to create both a Node.js instance and
web browser instance for when `--inspect` is used in Node.js (#483).

### CODE REVIEW

**Comment by StreetStrider** [comment]
@TooTallNate both `6ab9525c9841b656d996e521cf86192d5647483a` and current npm latest works well with Rollup now. 🤔

File: src/common.js [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+   * @param {String} namespace
+   * @return {Number}
+   * @api private
+   */
+
+  function selectColor(namespace) {
+    var hash = 0, i;
+
+    for (i in namespace) {
+      hash  = ((hash << 5) - hash) + namespace.charCodeAt(i);
```
Comment (thebigredgeek) [inline src/common.js]: Woah haha!

**Comment by homerjam** [comment]
Hey! Is this going to get released soon? 🙏

**Comment by bitflower** [comment]
Yeah, we need this also in StencilJS ! ;-)

**Comment by dmnsgn** [comment]
Hi @TooTallNate , could we get this published to npm? It seems that 3.1.0 doesn't have this and rollup isn't working without this fix. Thanks!

**Pull request merged**